### ISSUE TITLE
Commands are not found in new sessions

### ISSUE THREAD

**Issue by amir20** [fisherman/fisherman#214]
I can't figure out why this is happening and I am tried of debugging. Very often, after updating, (or some action) my commands are no longer found. For example, `z` is clearly installed by fish says `unknown command`. 

<img width="956" alt="screen shot 2016-04-22 at 12 51 28 pm" src="https://cloud.githubusercontent.com/assets/260667/14753265/0d397e1c-0889-11e6-914a-94f2040178dd.png">

**Comment by amir20** [comment]
So after a while, I noticed that this happens when I open a new tab. Has any body else found this to be a problem? Because it sounds like a big bug.

**Comment by amir20** [comment]
@bucaran Looks like whatever that was committed in the last day broke this. I confirmed with a new install that `fisher package` works after installation. But opening a new window doesn't.

**Comment by portenez** [comment]
+1 I also have this problem with z

**Comment by amir20** [comment]
It has also broke `pyenv`, `rbenv`.  But not `bass`. Which is weird.

**Comment by bucaran** [comment]
@amir20 @portenez 

**tl;dr**: if you are using the latest fish this is already taken care for you and it will be so in 2.3.0.

> [fisherman/fisherman#1-what-fish-version-is-required](https://github.com/fisherman/fisherman#1-what-fish-version-is-required)
fisherman was built for the latest fish, but at least 2.2.0 is required. If you can't upgrade your build, append the following code to your ~/.config/fish/config.fish for snippet support.
>
```fish
for file in ~/.config/fish/conf.d/*.fish
    source $file
end
```

**Comment by amir20** [comment]
@bucaran I have version 2.2.0. Do you think it's a good experience for fisherman to NOT work with latest stable version? I only ask because the documentation says latest version of fish. Which is what is installed. So it should work, but really you mean latest HEAD which is something different. 

I'd say update the documentation.

**Comment by bucaran** [comment]
@amir20 I am confused, fisherman *does* work with 2.2.0, in fact, that's the version I use.

<img width="408" alt="screen shot 2016-04-23 at 1 14 02 pm" src="https://cloud.githubusercontent.com/assets/8317250/14759215/42ec2cf4-0955-11e6-9359-e0629f577e34.png">

**Comment by amir20** [comment]
Then I don't think you understood this bug. I am saying that I see this but with the latest version of fish. Is that to be expected? So I guess we should reopen this ticket because I wasn't running an older version. Sorry if that wasn't clear.  I have the latest version via brew on Mac OS.

**Comment by bucaran** [comment]
@amir20 This is not a bug. In order to run "snippets", that is, .fish files which are evaluated / sourced on shell startup, you need fish from HEAD or add to your config.fish the following:

```fish
for file in ~/.config/fish/conf.d/*.fish
    source $file
end
```

It says so in the faq: https://github.com/fisherman/fisherman#1-what-fish-version-is-required

Now, we could definitely rewrite / reword some of that.

**Comment by herrbischoff** [comment]
@amir20: Please post the output of `fish -v` so we know exactly what version of fish you are using.

**Comment by bucaran** [comment]
@herrbischoff He is using 2.2.0.

**Comment by herrbischoff** [comment]
@bucaran: Just checking to cover all bases.

**Comment by bucaran** [comment]
@amir20 Okay, I just re-read this report, and I am starting to think the

**Comment by amir20** [comment]
I am away this weekend so I won't be able to give you the exact version. I am responding with my phone. I can get you the correct version on Sunday. But it's the latest 2.2 version on Mac via brew. 

Maybe @portenez can share his version? I just know this started happening after I upgraded to the latest fisherman.

**Comment by bucaran** [comment]
@amir20 I know you are busy away this weekend, but I feel you are not hearing my words, which is almost disconcerting. 

Can you read this: https://github.com/fisherman/fisherman#1-what-fish-version-is-required? 

Thanks.

**Comment by bucaran** [comment]
#### Some commands are not found in new sessions. Is this a bug in fisherman?

**No**. If you are running fish 2.2.0 (you probably are), you need to add the following code to your `~/.config/fish/config.fish` for snippet support.

```fish
for file in ~/.config/fish/conf.d/*.fish
    source $file
end
```

#### Okay, now everything works. What is a snippet anyway?

"Snippets" is a new feature in fish 2.3.0 that sources all *.fish* files inside a new directory named `conf.d` in your configuration (`~/.config/fish/conf.d`) when the session starts (e.g., open up a new window/tab, `exec fish < /dev/tty`, etc).

Snippet plugins like z, pyenv, rbenv, rvm, etc., depend on this behavior to function properly.

### I see, why didn't you say so in the first place?

I [did](https://github.com/fisherman/fisherman#1-what-fish-version-is-required).

**Comment by amir20** [comment]
What am I missing? That faq says you must have the latest version installed. It requires minimum 2.2.0 and if you don't have latest then you must add this snippet of code. 

Ok so I have latest installed. According to fish website 2.2.0 is the latest version. 

Reading your documentation suggest I should be able install using brew. And it should work. That's unclear. I am not trying to be annoying. I just think your documentation doesn't say that. 

Edit: the documentation should say versions < 2.3 requires this snippet.

**Comment by bucaran** [comment]
@amir20 Good point. The latest fish accordin to the website is 2.2.0, but that's going to change very very soon:

https://github.com/fish-shell/fish-shell/releases

**Comment by bucaran** [comment]
Improved wording in the question, I think it's much clearer now. Thanks!

https://github.com/fisherman/fisherman#1-what-fish-version-is-required

**Comment by bucaran** [comment]
@amir20 Good point.

Improved wording in the question, I think it's much clearer now. Thanks!

https://github.com/fisherman/fisherman#1-what-fish-version-is-required

**Comment by amir20** [comment]
Much better!

**Comment by bucaran** [comment]
Yeah, agreed the previous wording was misleading. Thanks for bearing with me haha.

**Comment by portenez** [comment]
Guys I'm using 2.2.0. I think this is going to be a very popular version because it's the one provided by homebrew... at least for the time being.

**Comment by bucaran** [comment]
@portenez 2.3.0 will be released very soon, days? weeks? The pre-release is already out btw:
https://github.com/fish-shell/fish-shell/releases/tag/2.3b1

For the time being, just add this to your config.fish


```fish
for file in ~/.config/fish/conf.d/*.fish
    source $file
end
```

>  provided by homebrew

I am not a brew ninja, but AFIK you can pull from HEAD using the cli, so even with brew you should be able to upgrade to the true latest.

**Comment by herrbischoff** [comment]
That would be `brew reinstall --HEAD fish`.

**Comment by denji** [comment]
`brew upgrade --HEAD fish`

**Comment by therealmarv** [comment]
the last command should be better in a fish shell

`brew up; and brew upgrade --HEAD fish;`

**Comment by amir20** [comment]
I can confirm that `brew reinstall --HEAD fish` fixes the problem.  @bucaran Thanks for understanding and fixing the documentation. I was also confused about the doc and I am not sure when `v2.3.0` will be released. There is nothing about official release data except [staging](http://fishshell.com/staging/) which only talks about a prerelease.  Hopefully it will be released soon because I can imagine a lot of people finding this confusing.

**Comment by bucaran** [comment]
@amir20 So you are using 2.2.0 with the `config.fish` snippet support code as described in #faq#1 or just 2.3.0?

**Comment by amir20** [comment]
I updated to `HEAD` which I believe is not `2.3.0` but it works. When I do `fish -v` I see `fish, version 2.2.0-793-gdcef1a5`. I chose this because I prefer minimal changes from the vanilla install. 👍

**Comment by bucaran** [comment]
Cool.
